### PR TITLE
workflow-state command back compat.

### PR DESCRIPTION
**Pull request by hjoliver**
Address #5236 
Sibling of #5238 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/scripts/workflow_state.py [hunk 1]
```diff
@@ -115,7 +115,10 @@ def connect(self):
             sys.stderr.write('\n')
 
         if connected and self.args['cycle']:
-            fmt = self.checker.get_remote_point_format()
+            try:
+                fmt = self.checker.get_remote_point_format()
+            except sqlite3.OperationalError:
+                fmt = self.checker.get_remote_point_format_back_compat()
```
Comment (MetRonnie) [inline cylc/flow/scripts/workflow_state.py:121]: Perhaps
```suggestion
            except sqlite3.OperationalError as exc:
                try:
                    fmt = self.checker.get_remote_point_format_back_compat()
                except sqlite3.OperationalError:
                    raise exc  # original error
```

File: cylc/flow/rundb.py [hunk 2]
```diff
@@ -179,6 +179,7 @@ class CylcWorkflowDAO:
     TABLE_BROADCAST_STATES = "broadcast_states"
     TABLE_INHERITANCE = "inheritance"
     TABLE_WORKFLOW_PARAMS = "workflow_params"
+    TABLE_SUITE_PARAMS = "suite_params"  # back compat
```
Comment (MetRonnie) [inline cylc/flow/rundb.py:182]: Consider moving this to end and adding a comment like
```python
# BACK COMPAT: suite_params
# This Cylc 7 DB table is needed to allow workflow-state
# xtriggers (and the `cylc workflow-state` command) to
# work with Cylc 7 workflows.
# url: https://github.com/cylc/cylc-flow/issues/5236
# remove at: 8.x
TABLE_SUITE_PARAMS = "suite_params"
```

**Comment by hjoliver** [comment]
Codecov patch fail is because I'm not testing the `cylc workflow-state` script directly. I've added a unit test for the underlying functionality though; I think that's sufficient in this case (making a functional test for the script, with old and new DBs, seems more painful than it's worth).

File: cylc/flow/rundb.py (hunk 2 above)
Comment (hjoliver) [reply cylc/flow/rundb.py]: Done.

**Comment by hjoliver** [comment]
(Reviewers: last commit is comment-only, so skip-ci added. All tests passed previously except slightly low on patch coverage as per my previous comment).

**Comment by hjoliver** [comment]
Actually, I'll re-base this to 8.x. It's arguably a bug fix, and needed ASAP at NIWA.

**Review by MetRonnie — APPROVED** [review]
LGTM but haven't tested

**Review by dwsutherland — APPROVED** [review]
Changes look good to me 👍 
(I would have to build an older VM or use work machines to test)

**Comment by dwsutherland** [comment]
The sibling is probably equally important (as some cylc8 workflows may be upstream of cylc7 on migration).

**Review by MetRonnie — CHANGES_REQUESTED** [review]
Ok managed to test out but it's not working. And the error message only shows up in `--debug` 

```
DEBUG - [xtrigger-func cmd] cylc function-run workflow_state '[]' '{"workflow": "cylc7-complex", "task": "brief_wagtail_forecast_landsim", "point": "20150219T0000Z"}' ~/cylc-
    run/workflow-state
    [xtrigger-func ret_code] 1
    [xtrigger-func err]
    Traceback (most recent call last):
      File "~/.conda/envs/cylc8/bin/cylc", line 8, in <module>
        sys.exit(main())
      File "~/cylc-flow/cylc/flow/scripts/cylc.py", line 660, in main
        execute_cmd(command, *cmd_args)
      File "~/cylc-flow/cylc/flow/scripts/cylc.py", line 286, in execute_cmd
        entry_point.resolve()(*args)
      File "~/cylc-flow/cylc/flow/scripts/function_run.py", line 41, in main
        run_function(args[1], args[2], args[3], args[4])
      File "~/cylc-flow/cylc/flow/subprocpool.py", line 115, in run_function
        res = func(*func_args, **func_kwargs)
      File "~/cylc-flow/cylc/flow/xtriggers/workflow_state.py", line 93, in workflow_state
        fmt = checker.get_remote_point_format()
      File "~/cylc-flow/cylc/flow/dbstatecheck.py", line 80, in get_remote_point_format
        for row in self.conn.execute(
    sqlite3.OperationalError: no such table: workflow_params
```

**Review by MetRonnie — COMMENTED** [review]
I've pushed a fix. May be sensible to add a functional test with a minimal Cylc 7 workflow DB, I can have a look at doing this if you're short on time

**Comment by hjoliver** [comment]
> I've pushed a fix. May be sensible to add a functional test with a minimal Cylc 7 workflow DB, I can have a look at doing this if you're short on time

Argh, last minute non-careful function name change by me, damn it. Thanks for the fix.

**Review by MetRonnie — APPROVED** [review]

**Comment by MetRonnie** [comment]
@hjoliver Do you want to merge if happy with the test I've added

**Comment by hjoliver** [comment]
Yes, very happy - thanks @MetRonnie

**Pull request merged**